pkg/cmd/server/origin/run_components.go

```diff
@@ -376,7 +376,7 @@ func (c *MasterConfig) RunDeploymentImageChangeTriggerController() {
 // RunSDNController runs openshift-sdn if the said network plugin is provided
 func (c *MasterConfig) RunSDNController() {
 	oClient, kClient := c.SDNControllerClients()
-	controller, err := sdnfactory.NewMasterPlugin(c.Options.NetworkConfig.NetworkPluginName, oClient, kClient)
+	controller, err := sdnfactory.NewMasterPlugin(c.Options.NetworkConfig.NetworkPluginName, oClient, kClient, c.EtcdHelper)
```

Comment:
I see these issues with the suggested approach:
- No change to clients: This is undesirable
  (1) Old client calls are synchronous and to ensure backward compatibility, server needs to poll to get assigned vnid for namespace (we want this to happen on the client side).
  (2) This introduces unnecessary indirection and complicates feature addition. Upcoming features like isolate-projects (undo join-projects) and ingress policy(upstream) stores info as annotations on namespace. To support this functionality via pod-network cli cmd in this case will need to pass annotation to NetNamespace first and then propagate to Namespace. 
- I guess migrator is expected to run after upgrade of all nodes and masters. Once migrator is called, it will perform copy to namespace and deletes object but any create calls after that will still create etcd based NetNamespace objects? How long we will maintain this?

NetNamespace is very specific to multitenant sdn plugin (unfortunately we didn't enable this resource conditionally like builds). This was exposed to support vnid watches on nodes and controlled updates by pod-network CLI. We don't expect/want admin users to explicitly create/delete this resource, if they do then there could be problems with isolation.

If we don't want controller to have direct access to etcd then we could do:
- By default, make NetNamespace act as virutal resource backed by namespace. Old client works without any issues. Change List/Update/Delete methods to handle custom flag and when this flag is enabled, it should list/delete etcd backed netnamespaces. This will be used for migration.
- New client hits Namespaces, there is no dependency on obsoleted NetNamespace.
- New nodes watches for both NetNamespaces and Namespaces, so new nodes can work with old masters. 
- New master will migrate NetNamespaces (client using custom flag) [Migration order: nodes and then  masters]
This will be much cleaner for future changes. No down time during migration, maintains backward compatibility and don't need to deal with etcd backed NetNamespaces after migration (no dual sets of values from namespaces and netnamespaces).

For (b), Agree VNIDAllocationController/SecurityAllocationController doesn't need etcd backing, I will change that to use in memory allocation.

@knobunc @dcbw @danwinship @rajatchopra 
@liggitt @smarterclayton any objections?